### PR TITLE
Add docker-compose

### PR DESCRIPTION
**Pull request by GosuOoops**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: "3.4"
+services:
+  docker:
+    image: homework25:latest
+    container_name: docker-compose
+    build: /home/authz/git
+    ports:
+      - 80:80
```
Comment (ggramal) [inline docker-compose.yaml:8]: Everything was installed just fine. But when i connect to the host and do a
```
$ curl localhost
curl: (56) Recv failure: Connection reset by peer
```
as you see there is an error :thinking:

Comment (GosuOoops) [reply docker-compose.yaml]: checked, found that flask works by default on 5000 port

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+version: "3.4"
+services:
+  docker:
+    image: homework25:latest
+    container_name: docker-compose
+    build: /home/authz/git
```
Comment (ggramal) [inline docker-compose.yaml:6]: lets use relative paths here. In order for developers to be able to run this service locally

File: ansible/playbook.yml [hunk 3]
```diff
@@ -2,4 +2,5 @@
   roles:
     - createuser
     - apps
```
Comment (ggramal) [inline ansible/playbook.yml:4]: Do we still need to install python with docker?

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+version: "3.4"
+services:
+  docker:
+    image: homework25:latest
+    container_name: docker-compose
+    build:
```
Comment (ggramal) [inline docker-compose.yaml:6]: do we need to use `image:` and `build:` at the same time :thinking:

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+      context: .
+      dockerfile: Dockerfile
+    expose:
+      - 5000
```
Comment (ggramal) [inline docker-compose.yaml:10]: seems like we are not forwarding any how ports in this compose
```
$ ss -tlnp
State            Recv-Q           Send-Q                     Local Address:Port                      Peer Address:Port           Process           
LISTEN           0                4096                       127.0.0.53%lo:53                             0.0.0.0:*                                
LISTEN           0                128                              0.0.0.0:22                             0.0.0.0:*                                
LISTEN           0                4096                           127.0.0.1:35021                          0.0.0.0:*                                
LISTEN           0                128                                 [::]:22                                [::]:*                                
```

File: docker-compose.yaml (hunk 4 above)
Comment (GosuOoops) [reply docker-compose.yaml:6]: today I just studied 26 homework, and saw when using image, this is just already a manual assembly, then build is not needed

File: docker-compose.yaml (hunk 5 above)
Comment (GosuOoops) [reply docker-compose.yaml]: fixed, try

File: ansible/roles/docker-compose/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  become: true
+  user:
+    name: authz
+    groups: docker
+    append: yes
+
+- name: Run container
+  become: true
+  command:
+    cmd: "docker-compose -f docker-compose.yaml up -d"
```
[the remark this answers is not in the document]
Comment (GosuOoops) [reply ansible/roles/docker-compose/tasks/main.yml:32]: fixed, try

[the remark this answers is not in the document]
Comment (ggramal) [reply ansible/roles/docker-compose/tasks/main.yml:32]: Still error
```
 {"changed": true, "cmd": ["docker-compose", "-f", "docker-compose.yaml", "up", "-d"], "delta": "0:00:01.604066", "end": "2023-07-06 10:29:11.062890", "msg": "non-zero return code", "rc": 1, "start": "2023-07-06 10:29:09.458824", "stderr": "Pulling docker (homework25:latest)...\nThe image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.\n\npull access denied for homework25, repository does not exist or may require 'docker login': denied: requested access to the resource is denied", "stderr_lines": ["Pulling docker (homework25:latest)...", "The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.", "", "pull access denied for homework25, repository does not exist or may require 'docker login': denied: requested access to the resource is denied"], "stdout": "Continue with the new image? [yN]", "stdout_lines": ["Continue with the new image? [yN]"]}

```

**Review by ggramal — APPROVED** [review]

**Pull request merged**